src/components/views/location/Map.tsx

```diff
@@ -91,26 +91,31 @@ const useMapWithStyle = ({
         }
     }, [map, bounds]);
 
-    const [geolocate] = useState(
-        allowGeolocate
-            ? new maplibregl.GeolocateControl({
-                  positionOptions: {
-                      enableHighAccuracy: true,
-                  },
-                  trackUserLocation: false,
-              })
-            : null,
-    );
+    const [geolocate, setGeolocate] = useState<maplibregl.GeolocateControl | null>(null);
 
     useEffect(() => {
-        if (map && geolocate) {
+        if (!map) {
+            return;
+        }
+        if (allowGeolocate && !geolocate) {
+            const geolocate = new maplibregl.GeolocateControl({
+                positionOptions: {
+                    enableHighAccuracy: true,
+                },
+                trackUserLocation: false,
+            });
+            setGeolocate(geolocate);
             map.addControl(geolocate);
             geolocate.on("error", onGeolocateError);
             return () => {
```

Comment:
I think the `"logs and opens a dialog on a geolocation error"` test is failing because this cleanup function is running immediately after geolocate is set, because the effect is triggered by the change to `geolocate` in state.

Reply:
Possibly putting the geo error handling part in a separate effect that only watches `geolocate` would fix it?
```
geolocate.on("error", onGeolocateError);
return () => { 
            geolocate.off("error", onGeolocateError);
};
```

Reply:
Oh, of course! 🤦‍♂️

Thanks, I changed it like that.